.github/workflows/windows.yml

```
@@ -32,17 +40,44 @@ jobs:
         sed -i \
           "s/^version =\".*\"/version =\"${version}\"/" \
           pyproject.toml
-    
-    - name: Install dependencies
-      run: make ci-install && pip install '.[windows]'
-      shell: bash
+        msi_version=$(
+            git describe --tags --dirty \
+            | sed -e 's/^v//' -e 's/-\([^-]*-g\)/.\1/;s/-g.*//g'
+          )
+        echo "MSI_VERSION=${msi_version}" >> "$GITHUB_ENV"
+          
 
-    - name: Build Windows binary
+    - name: Build Windows binaries
       run: |
-        pyinstaller --clean scripts/windows/papis.spec
-    
+        pyinstaller --clean scripts/windows/papis_onefile.spec
+        pyinstaller --clean scripts/windows/papis_onedir.spec
+
     - name: Publish binary as artifact
       uses: actions/upload-artifact@v4
       with:
-        name: papis-${{env.PAPIS_VERSION}}
+        name: papis-${{env.PAPIS_VERSION}}.exe
         path: ${{ github.workspace }}/dist/papis.exe
+
+    - name: Add msbuild to PATH
+      uses: microsoft/setup-msbuild@v2
+
+    - name: Install WiX
+      run: |
+        dotnet tool install --global wix --version 4.0.5
+        wix extension -g add WixToolset.UI.wixext/4.0.5
+    
+    - name: Render the MSI build manifest template
+      shell: bash
+      run: |
+        python scripts/windows/wix-populate_internal_artifacts.py
```

```suggestion
        python scripts/windows/wix-populate-internal-artifacts.py
```
? Looks weird to me to mix dashes and underscores.

---

yeah I wanted to separate wix from the other stuff. this is fine too!